### PR TITLE
fix(at): throw error when using at to access undefined property

### LINKED ISSUE #404: at should throw if the key is missing
```
ask {
    const marta = { firstName: 'Marta' }
    marta:at('firstName3')
}
```
returns `null` (empty) but an error was expected (playground with AskVM v. 1.2.0)

### PR DESCRIPTION
**Pull request by dastin-sandura**
closes #404

### CODE REVIEW

**Comment by mhagmajer** [comment]
@dastin-sandura thanks for creating a PR, it looks nice! You have covered the object case we'll, buy what about the case where `at` is used on a list? Can we cover that as well and add a relevant unit test?

```
const list =[1,2,3]
list[3]
```
Should probably also throw. Note that the code above is in AskScript and after transpilation to AskCode it will use the `at` resource.

**Comment by dastin-sandura** [comment]
Sure @mhagmajer :) I will add a test and implementation for your case.
As I understand your requirement, It will work something like '[index out of bounds exception](https://docs.oracle.com/javase/7/docs/api/java/lang/IndexOutOfBoundsException.html)' in Java?

**Comment by mhagmajer** [comment]
@dastin-sandura yes, that would be great and consistent with the behavior you have already implemented 👍

**Comment by dastin-sandura** [comment]
I will add it and let you know when done 😃

**Comment by dastin-sandura** [comment]
@mhagmajer done and ready for review 😄

**Review by YonatanKra — APPROVED** [review]
LGTM - @mhagmajer ?

**Comment by dastin-sandura** [comment]
@mhagmajer could you make this pull request count towards Hacktoberfest 2020? (https://hacktoberfest.digitalocean.com/)

**Comment by dastin-sandura** [comment]
@mhagmajer I checked my Hacktoberfest account and I see that it already counts 👍

File: package-lock.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "askql",
-  "version": "1.2.1",
+  "version": "1.2.2",
```
Comment (mhagmajer) [inline package-lock.json:3]: I forgot to update in on the last release :+1:

File: src/askvm/resources/core/at.ts [hunk 2]
```diff
@@ -3,6 +3,27 @@ import { any, resource, typed } from '../../lib';
...
+      Array.isArray(listOrObject) &&
+      Number.isInteger(key) &&
+      (listOrObject.length <= key || key < 0)
+    )
+      throw Error(
+        'Sorry, but index ' +
+          key +
+          ' is out of bounds for a list with ' +
+          listOrObject.length +
+          ' elements.'
```
Comment (mhagmajer) [inline src/askvm/resources/core/at.ts:16]: Can we use [template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) here for greater visibility?

**Review by mhagmajer — CHANGES_REQUESTED** [review]
Just a nit to fix :)

File: src/askvm/resources/core/at.ts (hunk 2 above)
Comment (dastin-sandura) [reply src/askvm/resources/core/at.ts:16]: sure 🙂

Comment (dastin-sandura) [reply src/askvm/resources/core/at.ts]: @mhagmajer btw thanks for the direct link to documentation about template literals 😄

**Review by dastin-sandura — COMMENTED** [review]
Refactored using template literals 👍

**Review by mhagmajer — APPROVED** [review]
Thanks!

**Pull request merged**